### ISSUE TITLE
Laskutustyökalussa virhe

### ISSUE THREAD

**Issue by MikkoLiimatainen** [KohaSuomi/Koha#1031]
### Mikä vikana?

Finvoice-laskuja lähettäessä tuli vastaan seuraavanlainen virhe:

![image](https://github.com/KohaSuomi/Koha/assets/98884651/ac844833-a514-4254-8607-d8a0c8d6ba2a)

Uskoisimme ongelman johtuvan asiakkaalla [90833](https://vaski.koha-suomi.fi/cgi-bin/koha/members/moremember.pl?borrowernumber=90833) olevasta lainasta teokseen ["Tyttö sinä olet tähti".](https://vaski.koha-suomi.fi/cgi-bin/koha/catalogue/detail.pl?biblionumber=2704982) Tämän asiakkaan lasku jäi ainoana muodostumatta niistä Finvoice-laskuista, joita yritettiin lähettää.

Mitään muuta selvästi erikoista tässä ei pistänyt silmään kuin nuo lainausmerkit tietueen nimessä, jotka ehkä rikkovat XML:n.

Virheen takia työntekijämme lähetti muut laskut kahdesti, joten olisiko mahdollista peruuttaa viestit joiden message_id:t ovat:

5827512
5827511
5827510
5827509
5827508
5827507
5827506
5827505
5827504
5827503
5827502
5827501
5827499
5827498
5827500
5827497
5827496
5827495
5827494
5827493
5827492
5827490
5827491
5827488
5827489
5827487
5827486
5827484
5827482
5827483
5827485
5827481
5827480
5827479
5827478
5827474
5827475
5827476
5827477
5827473

**Comment by emta001** [comment]
Viestien status vaihdettu "failed" tilaan seuraavalla rimpsulla:

`update message_queue set status = "failed" where message_transport_type = "finvoice" and message_id between 5827473 and 5827512;`

**Comment by emta001** [comment]
Äkkiseltään nuo lainausmerkit on tässä varmaan se potentiaalisin syyllinen, mutta pitänee vielä testailla.

**Comment by emta001** [comment]
Joo, sain tämän toistettua omassa testiympäristössä. XML ei tosiaan pidä lainausmerkeistä :D

**Comment by johannaraisa** [comment]
Missä kentässä nuo lainausmerkit oli? Työkalun pitäisi muuttaa titlestä ja authorista nuo lainausmerkit xml:n hyväksymään muotoon?

**Comment by MikkoLiimatainen** [comment]
Lainausmerkit ovat titlessä. Minulla oli mielikuva, että tuollainen käsittely on olemassa, mutta tuo näytti potentiaalisimmalta ongelman aiheuttajalta.

**Comment by johannaraisa** [comment]
Pitää varmaan ottaa tuo tietue itselle ja katsoa mitä siellä tapahtuu.

**Comment by johannaraisa** [comment]
Näyttää siltä, että tuo jälkimmäinen muutettu lainausmerkki menee rikki, kun tuo menee Kohan viestinmuodostukseen. Jostain syystä se hävittää lainausmerkistä puolipisteen, &quot; -> &quot

**Comment by johannaraisa** [comment]
Tuo nyt näköjään pitää hoitaa jossain toisessa kohtaa, siivoavat ylimääräiset ISBD-merkit lopusta.

**Comment by johannaraisa** [comment]
Päivitän tämän liitännäisen huomenna, vaihdoin tuon käsittelyn paikkaa.

**Comment by johannaraisa** [comment]
Nyt on tuotannossa tuo käsittely vaihdettu muualle, pitäisi nyt laskun muodostus onnistua.

**Comment by MikkoLiimatainen** [comment]
Epäonnistunut lasku saatiin eilen muodostettua ja se oli tänään tarkistuslistassa mukana. Pieni bugi tuossa vielä oli, kun titlestä "Tyttö sinä olet tähti". puuttui jälkimmäinen lainausmerkki ja loppupiste. Loppupisteet titleistä varmaan poistetaan tarkoituksella, mutta poistuukohan tuo lainausmerkki nyt siinä samassa? Tämänkin kanssa varmaankin tulemme kyllä toimeen. En ainakaan nyt keksi, että milloin tuosta voisi olla oikeasti ongelmia.

**Comment by johannaraisa** [comment]
> Epäonnistunut lasku saatiin eilen muodostettua ja se oli tänään tarkistuslistassa mukana. Pieni bugi tuossa vielä oli, kun titlestä "Tyttö sinä olet tähti". puuttui jälkimmäinen lainausmerkki ja loppupiste. Loppupisteet titleistä varmaan poistetaan tarkoituksella, mutta poistuukohan tuo lainausmerkki nyt siinä samassa? Tämänkin kanssa varmaankin tulemme kyllä toimeen. En ainakaan nyt keksi, että milloin tuosta voisi olla oikeasti ongelmia.

En tuota huomannut, kun testasin. Sinne oli jäänyt aikaisemmasta merkkimääräpäivityksestä siivousrivi. Poistan tuon ja laitan seuraavaan päivitykseen.

**Comment by johannaraisa** [comment]
Tätä voi vielä kokeilla testillä.
